# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/tables/nsTableFrame.cpp
# Commit: 58733fee05e0
# Full Hash: 58733fee05e09a34891f992e7c71051e6581efbe
# Author: Ting-Yu Lin <tlin@mozilla.com>
# Date: 2022-10-12 09:45:09
# Regressor Bug: 1794456
# File Overlap Count: 1
# Description:
#   Bug 1794456 Part 2 - Remove DEBUG members in nsFrameList::Slice and nsFrameList::Enumerator. r=emilio
#   
#   They are set but never used.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D158985
# ==============================================================================

diff -r 7f0b494afb59 -r 58733fee05e0 layout/tables/nsTableFrame.cpp
--- a/layout/tables/nsTableFrame.cpp	Tue Oct 11 21:01:27 2022 +0000
+++ b/layout/tables/nsTableFrame.cpp	Tue Oct 11 21:01:27 2022 +0000
@@ -854,7 +854,7 @@
     TableArea damageArea(0, 0, 0, 0);
     bool shouldRecalculateIndex = !IsDeletedRowIndexRangesEmpty();
     if (shouldRecalculateIndex) {
-      ResetRowIndices(nsFrameList::Slice(mFrames, nullptr, nullptr));
+      ResetRowIndices(nsFrameList::Slice(nullptr, nullptr));
     }
     int32_t origNumRows = cellMap->GetRowCount();
     int32_t numNewRows = aRowFrames.Length();
@@ -2142,14 +2142,14 @@
       mColGroups.InsertFrame(this, lastColGroup, f);
       // Insert the colgroup and its cols into the table
       InsertColGroups(startColIndex,
-                      nsFrameList::Slice(mColGroups, f, f->GetNextSibling()));
+                      nsFrameList::Slice(f, f->GetNextSibling()));
     } else if (IsRowGroup(display->mDisplay)) {
       DrainSelfOverflowList();  // ensure the last frame is in mFrames
       // Append the new row group frame to the sibling chain
       mFrames.AppendFrame(nullptr, f);
 
       // insert the row group and its rows into the table
-      InsertRowGroups(nsFrameList::Slice(mFrames, f, nullptr));
+      InsertRowGroups(nsFrameList::Slice(f, nullptr));
     } else {
       // Nothing special to do, just add the frame to our child list
       MOZ_ASSERT_UNREACHABLE(
@@ -2394,7 +2394,7 @@
     if (cellMap) {
       cellMap->Synchronize(this);
       // Create an empty slice
-      ResetRowIndices(nsFrameList::Slice(mFrames, nullptr, nullptr));
+      ResetRowIndices(nsFrameList::Slice(nullptr, nullptr));
       TableArea damageArea;
       cellMap->RebuildConsideringCells(nullptr, nullptr, 0, 0, false,
                                        damageArea);
